crates/adapters/dydx/src/python/websocket.rs

```diff
@@ -0,0 +1,255 @@
+        pyo3_async_runtimes::tokio::future_into_py(py, async move {
+            let ticker = ticker_from_instrument_id(&instrument_id);
+            let sub = crate::websocket::messages::DydxSubscription {
+                op: crate::websocket::enums::DydxWsOperation::Subscribe,
+                channel: crate::websocket::enums::DydxWsChannel::Trades,
+                id: Some(ticker),
+            };
+            let payload = serde_json::to_string(&sub).map_err(to_pyvalue_err)?;
+            handler_cmd_tx
+                .send(crate::websocket::handler::HandlerCommand::SendText(payload))
+                .map_err(|e| to_pyvalue_err(anyhow::anyhow!("{}", e)))
+        })
+    }
+
+    /// Unsubscribes from public trade updates for a specific instrument.
+    #[pyo3(name = "unsubscribe_trades")]
+    fn py_unsubscribe_trades<'py>(
+        &self,
+        py: Python<'py>,
+        instrument_id: InstrumentId,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        let handler_cmd_tx = self.handler_cmd_tx.clone();
+        pyo3_async_runtimes::tokio::future_into_py(py, async move {
+            let ticker = ticker_from_instrument_id(&instrument_id);
+            let sub = crate::websocket::messages::DydxSubscription {
+                op: crate::websocket::enums::DydxWsOperation::Unsubscribe,
+                channel: crate::websocket::enums::DydxWsChannel::Trades,
+                id: Some(ticker),
+            };
+            let payload = serde_json::to_string(&sub).map_err(to_pyvalue_err)?;
+            handler_cmd_tx
+                .send(crate::websocket::handler::HandlerCommand::SendText(payload))
+                .map_err(|e| to_pyvalue_err(anyhow::anyhow!("{}", e)))
+        })
+    }
+
+    /// Subscribes to orderbook updates for a specific instrument.
+    #[pyo3(name = "subscribe_orderbook")]
+    fn py_subscribe_orderbook<'py>(
+        &self,
+        py: Python<'py>,
+        instrument_id: InstrumentId,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        let client = self.clone();
+        pyo3_async_runtimes::tokio::future_into_py(py, async move {
+            client
+                .subscribe_orderbook(instrument_id)
+                .await
+                .map_err(to_pyvalue_err)
+        })
+    }
+
+    /// Unsubscribes from orderbook updates for a specific instrument.
+    #[pyo3(name = "unsubscribe_orderbook")]
+    fn py_unsubscribe_orderbook<'py>(
+        &self,
+        py: Python<'py>,
+        instrument_id: InstrumentId,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        let client = self.clone();
+        pyo3_async_runtimes::tokio::future_into_py(py, async move {
+            client
+                .unsubscribe_orderbook(instrument_id)
+                .await
+                .map_err(to_pyvalue_err)
+        })
+    }
+
+    /// Subscribes to candle/kline updates for a specific instrument.
+    #[pyo3(name = "subscribe_candles")]
+    fn py_subscribe_candles<'py>(
```

Comment:
For the public API of these network clients, we should use the normalized Nautilus terminology when possible (so this should be `subscribe_bars`.

Other methods specific to this adapter can use their terminology, e.g. `subscribe_block_height` is OK.

Reply:
Done, thanks.